cmd/gardenlet/app/migration.go

```diff
@@ -165,3 +179,85 @@ func syncBackupSecretRefAndCredentialsRef(backup *gardencorev1beta1.SeedBackup)
 	// - both fields are set -> let the validation check if they are correct
 	// - credentialsRef refer to WorkloadIdentity -> secretRef should stay unset
 }
+
+func migrateMCMRBAC(ctx context.Context, seedClient client.Client) error {
+	namespaceList := &corev1.NamespaceList{}
+	if err := seedClient.List(ctx, namespaceList, client.MatchingLabels(map[string]string{v1beta1constants.GardenRole: v1beta1constants.GardenRoleShoot})); err != nil {
+		return err
+	}
+
+	var tasks []flow.TaskFn
+
+	for _, ns := range namespaceList.Items {
+		if ns.DeletionTimestamp != nil || ns.Status.Phase == corev1.NamespaceTerminating {
+			continue
+		}
+		namespace := ns
+		tasks = append(tasks, func(ctx context.Context) error {
+			clusterRoleBinding := &rbacv1.ClusterRoleBinding{}
+			if err := seedClient.Get(ctx, client.ObjectKey{Name: "machine-controller-manager-" + namespace.Name}, clusterRoleBinding); err != nil {
+				//If MCM clusterRoleBinding does not exist, nothing to do
+				if apierrors.IsNotFound(err) {
+					return nil
+				}
+				return err
+			}
+
+			mcm := machinecontrollermanager.New(seedClient, namespace.Name, nil, machinecontrollermanager.Values{})
+			err := mcm.DeployMigrate(ctx)
+			if err != nil {
+				return err
+			}
+
+			return nil
+		})
+	}
+
+	err := flow.Parallel(tasks...)(ctx)
+	if err != nil {
+		return err
+	}
+	_ = managedresources.DeleteForSeed(ctx, seedClient, "garden", "machine-controller-manager")
+	return nil
```

Comment:
Why do you ignore the error here?

```suggestion
	return managedresources.DeleteForSeed(ctx, seedClient, "garden", "machine-controller-manager")
```

Reply:
When I tried to set up gardener from zero, I saw that this statement errored out and it complained that the `MR` CRD was not present
In that case it seems fair that the error be ignored

Reply:
Rather tolerate the 'not found' error then instead of silently ignoring all kinds of errors?